core/lib/snapshots_applier/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+        };
+
+        recovery.recover_factory_deps(&mut storage).await?;
+
+        storage
+            .snapshot_recovery_dal()
+            .set_applied_snapshot_status(&recovery.applied_snapshot_status)
+            .await
+            .map_err(|err| Fatal(err.into()))?;
+
+        storage.commit().await.map_err(|err| Fatal(err.into()))?;
+
+        recovery.recover_storage_logs().await?;
+
+        Ok(())
+    }
+
+    async fn create_fresh_recovery_status(
+        main_node_client: Box<dyn SnapshotsApplierMainNodeClient>,
+    ) -> Result<SnapshotRecoveryStatus, SnapshotApplierError> {
+        let snapshot_response = main_node_client
+            .fetch_newest_snapshot()
+            .await
+            .map_err(|_| Retryable("Unable to fetch newest snapshot from main node".to_string()))?;
+        if snapshot_response.is_none() {
+            return Err(Canceled("Main node does not have any ready snapshots, skipping initialization from snapshot!".to_string()));
+        }
+        let snapshot = snapshot_response.unwrap();
+        let l1_batch_number = snapshot.l1_batch_number;
+        tracing::info!(
+            "Found snapshot with data up to l1_batch {}, storage_logs are divided into {} chunk(s)",
+            l1_batch_number,
+            snapshot.storage_logs_chunks.len()
+        );
+
+        let miniblock_root_hash = main_node_client
+            .miniblock_hash(snapshot.miniblock_number)
+            .await
+            .map_err(|err| Fatal(err.into()))?
+            .ok_or_else(|| Fatal(Error::msg("Miniblock is missing")))?;
+
+        let chunk_ids = snapshot
+            .storage_logs_chunks
+            .iter()
+            .map(|x| x.chunk_id)
+            .collect();
+        Ok(SnapshotRecoveryStatus {
+            l1_batch_number,
+            l1_batch_root_hash: snapshot.last_l1_batch_with_metadata.metadata.root_hash,
+            miniblock_number: snapshot.miniblock_number,
+            miniblock_root_hash,
+            storage_logs_chunks_ids_already_processed: vec![],
+            storage_logs_chunks_ids_to_process: chunk_ids,
+        })
+    }
+
+    async fn recover_factory_deps(
+        &mut self,
+        storage: &mut StorageProcessor<'_>,
+    ) -> anyhow::Result<(), SnapshotApplierError> {
+        let factory_deps: SnapshotFactoryDependencies = self
+            .blob_store
+            .get(self.applied_snapshot_status.l1_batch_number)
+            .await?;
+
+        let all_deps_hashmap: HashMap<H256, Vec<u8>> = factory_deps
+            .factory_deps
+            .into_iter()
+            .map(|dep| (hash_bytecode(&dep.bytecode.0), dep.bytecode.0))
+            .collect();
+        storage
+            .storage_dal()
+            .insert_factory_deps(
+                self.applied_snapshot_status.miniblock_number,
+                &all_deps_hashmap,
+            )
+            .await?;
+        Ok(())
+    }
+    async fn insert_initial_writes_chunk(
```

Comment:
Nit: Missing empty lines between methods (ditto in a couple of other places).

Reply:
I think I will just write a python script for my personal use that adds such lines as this is at least 3rd time I've missed this :D